code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -76,7 +77,7 @@
 		. += "[icon_door_override ? icon_door : icon_state]_open"
 		return
 
-	if(icon_door)
+	if(has_door_icon)
 		. += "[icon_door || icon_state]_door"
```

Comment:
It turns out that whoever added the icon_door variable didn't set it up on every closet and now the overlay isn't being added, can be checked on main.

Reply:
That is incorrect. The `cardboard_door` state for the cardboard box 'closet' does not (and should not) exist. The original `if(icon_door)` was added by SandPoot to mask this error, but it also broke anywhere that relied on an auto-set icon_state. This new `if(has_door_icon)` I added *disables adding the door icon entirely* for the cardboard box (and anything else that could possibly need it), rather than just clumsily skipping it if `icon_door` is unset.

SandPoot, the point of using `icon_door || icon_state` is to make it so that it doesn't have to be set on every closet. `icon_door` is specifically for closets whose door icon state differs than their base icon state. (This is not to be confused with `icon_door_override`, which is used when the *open* door icon state differs from the *closed* door icon state, I believe?)

Reply:
When an iconstate is missing from the DMI (such as for `cardboard_door`), Byond simply uses whatever iconstate in the icon has a null name. In the case of lockers, there's no iconstate with a null name, and thus `cardboard_door` will display nothing. Cardboard boxes should definitely be converted to the newer separated door format, though.

That being said, it feels a little overkill to have this check in place for a single edge-case, but the reasoning is fair enough, all things considered.

Reply:
The existing fix was worse, and variables with their default value don't take up any memory cost per-object. I'd rather have overkill that works than overkill that doesn't.